source/application/REIXS/REIXSAppController.cpp

```diff
@@ -78,85 +78,134 @@ REIXSAppController::REIXSAppController(QObject *parent) :
 }
 
 bool REIXSAppController::startup()
-{
-	bool success = true;
+{	
+	if(AMAppController::startup()) {
 
-	success &= AMAppController::startup();
-	success &= AMActionRegistry3::s()->registerInfoAndAction<AMSamplePlatePre2013MoveActionInfo, AMSamplePlatePre2013MoveAction>("Move Sample Position", "Move to a different marked sample position", ":system-run.png");
-	success &= AMActionRegistry3::s()->registerInfoAndEditor<AMSamplePlatePre2013MoveActionInfo, AMSamplePlatePre2013MoveActionEditor>();
+		// Initialize the central beamline object
+		REIXSBeamline::bl();
+		// Initialize the storage ring.
+		CLSStorageRing::sr1();
 
-	return success;
-}
 
-// Re-implemented to create the REIXSBeamline object
-bool REIXSAppController::startupBeforeAnything() {
-	if(!AMAppController::startupBeforeAnything()) return false;
+		registerClasses();
+
+		// Checking for and making the first run in the database, if there isn't one already.
+		////////////////////////////////////////
+		AMRun existingRun;
+		if(!existingRun.loadFromDb(AMDatabase::database("user"), 1)) {
+			// no run yet... let's create one.
+			AMRun firstRun(CLSFacilityID::beamlineName(CLSFacilityID::REIXSBeamline), CLSFacilityID::REIXSBeamline); //5: REIXS Beamline
+			firstRun.storeToDb(AMDatabase::database("user"));
+		}
 
-	if (!AMChooseDataFolderDialog::getDataFolder("/AcquamanLocalData/reixs", "/home/reixs", "users"))
+		setupUserInterface();
+		makeConnections();
+	} else {
 		return false;
+	}
+}
 
-	// Initialize the central beamline object
-	REIXSBeamline::bl();
-	// Initialize the storage ring.
-	CLSStorageRing::sr1();
+void REIXSAppController::shutdown() {
 
-	connect(REIXSBeamline::bl()->spectrometer(), SIGNAL(connected(bool)), REIXSBeamline::bl()->spectrometer(), SLOT(updateGrating()));
+	// Make sure we release/clean-up the beamline interface
+	AMBeamline::releaseBl();
```

Comment:
Hmm, interesting.  I think this must have slipped through the cracks because beamlines instantiate CLSStorageRing and CLS classes don't belong in AM classes (like AMAppController).  However, since the releaseStorageRing method is general anyway, you should be able to use ```AMStorageRing::releaseStorageRing()``` from ```AMAppController::shutdown()```.